pyard/pyard.py

```diff
@@ -429,6 +433,42 @@ def redux_gl(self, glstring: str, redux_type: str) -> str:
                 return self.redux_gl("/".join(sorted(alleles, key=functools.cmp_to_key(loci_sort))), redux_type)
         return self.redux(glstring, redux_type)
 
+    def isvalid(self, allele: str) -> str:
+        """
+        Determines valididy of an allele
+
+        :param allele: An HLA allele.
+        :type: str
+        :return: allele or empty
+        :rtype: boolean
+        """
+        v = lambda a: a in self.valid
+        return v(allele)
+
+    def isvalid_gl(self, glstring: str) -> str:
+        """
+        Determine validity of glstring
+
+        :param glstring
+        :type: str
+        :return: result
+        :rtype: boolean
+        """
+        
+        if re.search("\^", glstring):
+            return(all(list(map(self.isvalid_gl,glstring.split("^")))))
```

Comment:
```suggestion
            return(all(map(self.isvalid_gl,glstring.split("^"))))
```
No need to convert it into `list`.